packages/wdio-utils/src/envDetector.ts

```diff
@@ -200,16 +200,23 @@ export function capabilitiesEnvironmentDetector (capabilities: Capabilities.Capa
  * @param  {Object}  requestedCapabilities
  * @return {Object}                         object with environment flags
  */
-export function sessionEnvironmentDetector ({ capabilities, requestedCapabilities }: { capabilities: Capabilities.DesiredCapabilities, requestedCapabilities: Capabilities.DesiredCapabilities | Capabilities.W3CCapabilities }) {
+export function sessionEnvironmentDetector({ capabilities, requestedCapabilities }:
+    { capabilities: Capabilities.DesiredCapabilities | Capabilities.W3CCapabilities, requestedCapabilities: Capabilities.DesiredCapabilities | Capabilities.W3CCapabilities }) {
+    let cap: Capabilities.DesiredCapabilities
+    if ('alwaysMatch' in capabilities) {
+        cap = capabilities.alwaysMatch
+    } else {
+        cap = capabilities
+    }
```

Comment:
```suggestion
    const cap: Capabilities.Capabilities = 'alwaysMatch' in capabilities
        ? capabilities.alwaysMatch
        : capabilities
```

Reply:
wow, your solution is much more concise, thank you